node/tessel-export.js

```diff
@@ -969,11 +971,21 @@ Tessel.SPI = function(params, port) {
 };
 
 Tessel.SPI.prototype.send = function(data, callback) {
-  this._port.cork();
+
   this.chipSelect.low();
-  this._port._tx(data, callback);
-  this.chipSelect.high();
-  this._port.uncork();
+
+  setTimeout(() => {
+    // Only cork if we have specified a target delay, otherwise delay increases
+    if (this.chipSelectDelay > 0) {
+      this._port.cork();
+    }
+    this._port._tx(data, callback);
+    this.chipSelect.high();
+    if (this.chipSelectDelay > 0) {
+      this._port.uncork();
+    }
+  }, this.chipSelectDelay);
```

Comment:
The `setTimeout` still has a cost—any other code called after `send` but before the end of the execution turn will now execute before any of this code is executed. I think this needs to be reconsidered